Add owner policy (#618)
FuelLabs/fuel-specs: src/tx-format/policy.md

- `max_fee > sum_inputs(tx, BASE_ASSET_ID) - sum_outputs(tx, BASE_ASSET_ID)`
- `max_fee < max_fee(tx, BASE_ASSET_ID, gas_price)`

## `Owner`

| name    | type     | description                                              |
|---------|----------|----------------------------------------------------------|
| `owner` | `uint64` | Index of the transaction input that we nominate as owner |

Transaction is invalid if:

> - `owner >= inputs.len()`

Reviewer: It should also be invalid if the input type doesn't have an owner field.

Ie messages or contract inputs.

If we wanted both coins and messages supported then it should be owner in the case of coin inputs, and receiver in the case or message inputs.

Author: Mhh okey who can take a decision on what we want to be valid ? Given my little experience I'm ok with both.

Reviewer: It should probably be both messages and coins, given that `coinsToSpend` can select either a coin or message for the base asset.

Reviewer: @xgreenx could you clarify what you mean by this? It's not obvious in the context of design specs.

Reviewer 2: I meant that the policy is invalid if the index from the owner's policy doesn't point to the `Message` or `Coin` input.

Author: > I meant that the policy is invalid if the index from the owner's policy doesn't point to the `Message` or `Coin` input.

This is something handle on the sway level right ?

Author: Updated spec and code in VM PR.

- `max_fee > sum_inputs(tx, BASE_ASSET_ID) - sum_outputs(tx, BASE_ASSET_ID)`
- `max_fee < max_fee(tx, BASE_ASSET_ID, gas_price)`

## `Owner`

| name    | type     | description                                              |
|---------|----------|----------------------------------------------------------|
| `owner` | `uint64` | Index of the transaction input that we nominate as owner |

Transaction is invalid if:

- `owner >= inputs.len()`

> - `inputs[owner] != InputType::Message && inputs[owner] != InputType::Coin` (owner need to points to an input which has an owner field)

Reviewer: ```suggestion
- `inputs[owner] != InputType::Message && inputs[owner] != InputType::Coin` (must point to an input type that supports ownership)
```
